flink-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/common/RMQConnectionConfig.java

```diff
@@ -506,16 +532,44 @@ public Builder setPrefetchCount(int prefetchCount) {
             return this;
         }
 
+        /**
+         * Enables setting the next message delivery timeout in the queueing consumer. Only
+         * applicable to the {@link RMQSource}. Set to 0 for unlimited (the consumer will be blocked
+         * until an element becomes available). Default is 30000.
+         *
+         * @param deliveryTimeout maximum wait time, in milliseconds, for the next message delivery
+         * @return the Builder
+         */
+        public Builder setDeliveryTimeout(int deliveryTimeout) {
+            this.deliveryTimeout = deliveryTimeout;
```

Comment:
Why doesn't this check whether deliveryTimeout is negative?

Reply:
It's already checked in the constructor (invoked in the build method), but it can be checked here as well to indicate invalid value where it's actually set. Added input validation in f90e2d2 - thanks

Reply:
I agree with @austince's comment.